### DISCUSSION TITLE
consumer & route ratelimit

### DISCUSSION THREAD

**Discussion by caibirdme** [apache/apisix#5493] [Q&A]
According to the docs, plugin will be invoked only once during a request lifetime. But there's a problem. As an apigw, it needs to protect  the upstream, so I should add a rate-limit plugin for each route. But when I want to limit some consumer to access this route, I need to add ratelimit plugin for that consumer. as well. So how to achieve that goal?

**Comment by tokers** [comment]
Configure the rate-limiting key to the consumer name, see https://apisix.apache.org/docs/apisix/plugins/limit-req/#how-to-enable-on-the-consumer for details.

**Reply by caibirdme** [reply]
In the case, just consumer limit-req config works. I want to make sure the qps of the /a/b is no more than 3000qps, and consumer A for /a/b is no more than 100qps and consumer B for /a/b is no more than 200qps

**Reply by tokers** [reply]
I know your needs, but currently, same plugin can only be run once in a route, so preparing a few rate-limiting buckets are not feasible, if you're interested in it, welcome to submit an email with your proposal to the dev@apisix.apache.org.

**Reply by juzhiyuan** [reply]
https://apisix.apache.org/docs/general/subscribe-guide

Here is the subscription guide 😄

**Comment by xzy1990** [comment]
I also encounter this problem, have you solved it now?

**Reply by juzhiyuan** [reply]
Hi, do you have ideas on this comment? https://github.com/apache/apisix/discussions/5493#discussioncomment-1638647

**Reply by xzy1990** [reply]
up to now , I have no idea.
